### PR TITLE
implement GET lesson by lecturer endpoint

### PR DESCRIPTION
**Pull request by ezgy**
This pull request contains the implementation regarding lesson creation: a GET endpoint /lesson/byLecturer. Issue regarding this is https://github.com/bounswe/bounswe2022group2/issues/239. Implementation will be tested using Postman.

After the implementation and testing stages are done, I will be preparing a documentation for the API regarding this GET endpoint.

NOTE: As this endpoint required the lesson model to have a lecturer, the model has also been modified. Create lesson endpoint was updated accordingly(see #238 )

### CODE REVIEW

**Review by egemenatikk — DISMISSED** [review]
Thank you for your work @ezgy. This endpoint includes necessary operations for getting lessons by lecturer. I see that you also did the necessary change in Lesson model and create a lesson endpoint.

One thing I can suggest for this endpoint is that maybe you can also import User model and check if a User with the given lecturer_id exists or not before finding Lessons. I am not sure if it is necessary but it can be more comprehensive, maybe. I am approving this PR since it seems good enough and error-free. If you wish, you can add the functionality I talked about above.

File: practice-app/server/src/api/controllers/lesson/get_lesson_by_lecturer.js [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 
 export default async (req, res) => {
   const lecturerId = req.query.lecturer;
   if (!lecturerId || !mongoose.isValidObjectId(lecturerId)) {
     return res.status(400).json({ "resultMessage": "Please provide a valid lecturer id." });
   }
 
+  const lecturer = await User.findOne({ _id: lecturerId})
+  .catch((err) => {
+      return res.status(404).json({ "resultMessage": err.message });;
```
Comment (egemenatikk) [inline practice-app/server/src/api/controllers/lesson/get_lesson_by_lecturer.js:13]: This part represents an internal error, so its error status code should be 500 if I'm not wrong.

**Comment by ezgy** [comment]
I did the necessary updates @egemenatikk can you review them again?

**Review by egemenatikk — APPROVED** [review]
Thank you, now I can't think anything else to add, I think it is ready to be merged.

**Pull request merged**